### PR TITLE
Downgrade to React 17

### PR DESCRIPTION
**Pull request by LaChope**

### CODE REVIEW

**Comment by LaChope** [comment]
@blcham [35c9cbd](https://github.com/kbss-cvut/s-forms/pull/243/commits/35c9cbd4e5fd8ad7a7fa15c534b71be8acb0614e) resolves #246 (I commited in this branch as I assume the goal is to have it for the record-manager-ui)

File: src/contexts/ConfigurationContext.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const defaultProps = {
   componentsOptions: {
     dateTimeAnswer: {
       dateFormat: "yyyy-MM-dd",
-      timeFormat: "HH:mm:ss",
+      timeFormat: "HH:mm",
```
Comment (blcham) [inline src/contexts/ConfigurationContext.js:14]: Do not understand why this changed.

File: package.json [hunk 2]
```diff
@@ -58,22 +58,23 @@
     "npm-packlist": "^7.0.2",
     "prop-types": "^15.7.2",
     "react-bootstrap": "1.0.1",
-    "react-datepicker": "^4.14.1",
-    "react-intl": "^6.5.1",
-    "react-loader-spinner": "^5.3.4",
+    "react-datepicker": "^4.2.1",
+    "react-intl": "^5.24.6",
+    "react-loader-spinner": "^5.1.0",
+    "react-select": "^3.1.0",
```
Comment (blcham) [inline package.json:64]: How do we use this dependency?

File: src/stories/assets/question/simple-single-tree-select.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
+{
+  "@id": "ma-vlastnika-section-777",
+  "@type": "http://onto.fel.cvut.cz/ontologies/documentation/question",
+  "http://onto.fel.cvut.cz/ontologies/form-layout/has-layout-class": "type-ahead",
+  "http://onto.fel.cvut.cz/ontologies/form/has-non-selectable-value": "fyzicka-osoba",
+  "http://onto.fel.cvut.cz/ontologies/form/has-possible-value": [
+    {
+      "@id": "fyzicka-osoba",
+      "http://www.w3.org/2000/01/rdf-schema#label": "Fyzická osoba",
+      "http://www.w3.org/2002/07/owl#disjointWith": {
```
Comment (blcham) [inline src/stories/assets/question/simple-single-tree-select.json:10]: It does not make sense to have disjoint values

Comment (blcham) [reply src/stories/assets/question/simple-single-tree-select.json:10]: remove all of those references in this file

File: .github/workflows/npm-publish-alpha.yml [hunk 4]
```diff
@@ -34,7 +34,6 @@ jobs:
           node-version: 16
           registry-url: https://registry.npmjs.org
       - run: npm ci
-      - run: npm test
```
Comment (blcham) [inline .github/workflows/npm-publish-alpha.yml:37]: If we want to merge i guess it does not make sense to remove tests.

**Review by blcham — CHANGES_REQUESTED** [review]

**Comment by LaChope** [comment]
@blcham you can check the last 2 commits, https://github.com/kbss-cvut/record-manager-ui/issues/28 should be fixed now

File: src/stories/assets/question/simple-single-tree-select.json (hunk 3 above)
Comment (LaChope) [reply src/stories/assets/question/simple-single-tree-select.json:10]: Done

File: src/components/answer/TypeaheadAnswer.jsx [hunk 5]
```diff
@@ -73,37 +73,99 @@ const TypeaheadAnswer = (props) => {
...
+      let label = JsonLdUtils.getLocalized(
+        pValue[Constants.RDFS_LABEL],
+        configurationContext.options.intl
+      );
+
+      options[pValue["@id"]] = {
+        value: pValue["@id"],
+        label: label,
+        children: [],
+        disjoint: [],
```
[the remark this answers is not in the document]
Comment (LaChope) [reply src/components/answer/TypeaheadAnswer.jsx:99]: Done

File: package.json (hunk 2 above)
Comment (LaChope) [reply package.json:64]: I just bumped the packages

File: .github/workflows/npm-publish-alpha.yml (hunk 4 above)
Comment (LaChope) [reply .github/workflows/npm-publish-alpha.yml:37]: Tests are broken due to downgrading to react 17, we will not merge this branch but rather cherry pick

**Pull request merged**